xmppserver/src/main/java/org/jivesoftware/openfire/disco/IQDiscoInfoHandler.java

```diff
@@ -149,7 +149,7 @@ public IQ handleIQ(IQ packet) {
         // a <service-unavailable/> error if [...] The requesting entity is not authorized to receive presence from the
         // target entity (i.e., via the target having a presence subscription to the requesting entity of type "both" or
         // "from") or is not otherwise trusted (e.g., another server in a trusted network).</blockquote>
-        if (packet.getTo() != null && packet.getTo().getNode() != null && packet.getTo().getResource() == null) {
+        if (packet.getTo() != null && packet.getTo().getNode() != null && packet.getTo().getResource() == null && !packet.getTo().asBareJID().equals(packet.getTo().asBareJID())) {
```

Comment:
This seems odd to me `packet.getTo().asBareJID() == packet.getTo().asBareJID()` ?  How does this comparison not always be equal?

Reply:
:facepalm: One should've been 'from'. Fixed.